Lib/ufo2ft/outlineCompiler.py

```diff
@@ -1070,6 +1070,63 @@ def _bboxHeight(self, glyph):
         return bbox.yMax - bbox.yMin
 
     def setupTable_MATH(self):
+        """
+        This builds MATH table based on data in the UFO font. The data is stored
+        either in private font/glyph lib keys or as glyph anchors with specific
+        names.
+
+        The data is based on GlyphsApp MATH plugin data as written out by
+        glyphsLib to the UFO font.
+
+        The font lib keys are:
+            - com.nagwa.MATHPlugin.constants: a dictionary of MATH constants as
+              expected by fontTools.otlLib.builder.buildMathTable(). Example:
+
+                ufo.lib["com.nagwa.MATHPlugin.constants"] = {
+                    "ScriptPercentScaleDown": 70,
+                    "ScriptScriptPercentScaleDown": 60,
+                    ...
+                }
+
+            - com.nagwa.MATHPlugin.variants: a dictionary of MATH glyph variants
+              keyed by glyph names, and each value is a dictionary with keys
+              "hVariants" and "vVariants". Example:
+
+                ufo.lib["com.nagwa.MATHPlugin.variants"] = {
+                    "parenleft": { "vVariants": ["parenleft", "parenleft.s1", "parenleft.s2"] },
+                    "tildecomb": { "hVariants": ["tildecomb", "tildecomb.s1", "tildecomb.s2"] },
+                    ...
+                }
+
+            - com.nagwa.MATHPlugin.extendedShape: a list of glyph names that
+              are extended shapes. Example:
+
+                ufo.lib["com.nagwa.MATHPlugin.extendedShapes"] = ["integral", "radical"]
+
+        The glyph lib keys are:
+            - com.nagwa.MATHPlugin.variants: a dictionary of MATH glyph variants
+              keyed by glyph names, and each value is a dictionary with keys
+              "hAssembly" and "vAssembly". Example:
+
+                ufo["braceleft"].lib["com.nagwa.MATHPlugin.variants"] = {
+                    "vAssembly": [
+                        # glyph name, flags, start connector length, end connector length
+                        ["braceleft.bottom", 0, 0, 200],
+                        ["braceleft.extender", 1, 200, 200],
+                        ["braceleft.middle", 0, 100, 100],
+                        ["braceleft.extender", 1, 200, 200],
+                        ["braceleft.top", 0, 200, 0],
+                    ],
+                }
+
+        The anchors are:
+            - math.ic: italic correction anchor
+            - math.ta: top accent attachment anchor
+            - math.tr*: top right kerning anchors
+            - math.tl*: top left kerning anchors
+            - math.br*: bottom right kerning anchors
+            - math.bl*: bottom left kerning anchors
```

Comment:
is that a literal asterisk '*' or glob-style pattern?

Reply:
A glob. All anchors starting with "math.tr" etc. are used to build the math kerning (cut-ins) array of the relevant side:

![image](https://github.com/googlefonts/ufo2ft/assets/93914/8977ad04-5db3-406c-a683-388adf604287)

Reply:
ok.

I wonder, MATH table is not variable as far as I know. Are these pieces of data only meant to be stored in (and built from) the default UFO master?
Or maybe this is only meant to work for projects where the static instances and the masters have the same coordinates so no actual interpolation takes places and the instances inherit the data from the respective master?